src/SleighInstruction.cpp

```diff
@@ -2,39 +2,161 @@
 
 #include "SleighInstruction.h"
 
+SleighInstructionPrototype *R2Sleigh::getPrototype(SleighInstruction *context)
+{
+	SleighInstructionPrototype *new_proto = new SleighInstructionPrototype(this, context);
+	uint4 hash = new_proto->hashCode;
+
+	if(proto_cache.find(hash) == proto_cache.end())
+	{
+		new_proto->cacheTreeInfo();
+		proto_cache[hash] = new_proto;
+	}
+	else
+	{
+		delete new_proto;
+		new_proto = proto_cache[hash];
+	}
+
+	return new_proto;
+}
+
 SleighInstruction *R2Sleigh::getInstruction(Address &addr)
 {
-	SleighInstruction *ins = nullptr;
+	SleighInstruction *inst = nullptr;
+	/*
```

Comment:
Is this commented code still relevant? If not, please remove it.

Reply:
Keep this for potential usage. We can also cache SleighInstruction in future. More profile is needed.